resources/lib/plugin.py

```diff
@@ -179,6 +184,495 @@ def result_to_itemlist(result, playlist='', channel=''):
 
     return items
 
+def get_user_channel():
+    user_channel_str = ADDON.getSettingString('user_channel')
+    if user_channel_str:
+        toks = user_channel_str.split('#')
+        if len(toks) == 2:
+            return (toks[0], toks[1])
+        else:
+            ADDON.setSettingString('user_channel', '')
+    return None
+
+def set_user_channel(channel_name, channel_id):
+    ADDON.setSettingString('user_channel', "%s#%s" % (channel_name, channel_id))
+
+def clear_user_channel():
+    ADDON.setSettingString('user_channel', '')
+
+class CommentWindow(WindowXML):
+    def __init__(self, *args, **kwargs):
+        self.claim_id = kwargs['claim_id']
+        self.verified_user_channel = False
+        WindowXML.__init__(self, args, kwargs)
+
+    def onInit(self):
+        self.refresh()
+
+    def onAction(self, action):
+        if action == xbmcgui.ACTION_CONTEXT_MENU:
+            # Commenting is not supported
+            if using_lbry_proxy:
+                ret = dialog.contextmenu([tr(30240)]) # Only allow refreshing
+                if ret == 0:
+                    self.refresh()
+                return
+
+            # No user channel. Allow user to select an account or refresh.
+            if not get_user_channel():
```

Comment:
I don't like the user selection via the context menu of a comment, as it should be able to be deduced from the API.
(lbrynet channel list)

Reply:
select_user_channel() will automatically select the channel if there is only one channel. The user still has to manually click the item in the context menu to poll the server. With your proposed change an issue arises when the server has no accounts. In this case we would need to either:

-Poll the server each time the addon is started as there might be a user. The Lbry API servers seem to be fairly slow so this would introduce an fairly annoying hiccup with lbrynet servers that have no claimed channels.

-Store an extra bool in the settings stating that the server has no accounts. This approach would introduce an issue if an account is later added. We would need some way of allowing the server to be repolled for an account.

The context menu approach of selecting user accounts is perhaps not a classical UI design but its pragmatic and simplifies switching between accounts for users that post under different  channels. Adding more steps to change users when the context menu isn't cluttered seems unnecessary.

Reply:
Maybe I added this comment to the wrong location in the code, but the point I was trying to make is that the context menu on _every_ comment has a Select User Channel context menu option, which is confusing and can be solved in a better way I think.

> Poll the server each time the addon is started as there might be a user

No agreed, this sounds horrible

> Store an extra bool in the settings stating that the server has no accounts

Also not ideal. If possible, maybe just retrieve channels and select from the configuration screen

Reply:
Moved user selection to the settings page. There is a quirk where if the user changes the server they have to save the page, reopen, and then select the user. Kodi doesn't seem to save any settings until OK is pressed.

Reply:
Great work!

A quick test without any user channels (I have no channels/uploads) however throws an exception:

```
                                                   Traceback (most recent call last):
                                                     File "/home/sander/.kodi/addons/plugin.video.lbry/lbry.py", line 6, in <module>
                                                       plugin.run()
                                                     File "/home/sander/.kodi/addons/plugin.video.lbry/resources/lib/plugin.py", line 885, in run
                                                       plugin.run()
                                                     File "/home/sander/.kodi/addons/script.module.routing/lib/routing.py", line 130, in run
                                                       self._dispatch(self.path)
                                                     File "/home/sander/.kodi/addons/script.module.routing/lib/routing.py", line 141, in _dispatch
                                                       view_func(**kwargs)
                                                     File "/home/sander/.kodi/addons/plugin.video.lbry/resources/lib/plugin.py", line 232, in select_user_channel
                                                       progressDialog.update(int(100.0*page/total_pages), tr(30220) + ' %s/%s' % (page, total_pages))
                                                   ZeroDivisionError: float division by zero
                                                   -->End of Python script error report<--

```

Reply:
Fix submitted.